### ISSUE TITLE
如何在YOLO11的项目工程文件中用YOLOv8的模型训练与推理

### ISSUE THREAD

**Issue by 666git-sys** [ultralytics/ultralytics#20051]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

如何在YOLO11的项目工程文件中用YOLOv8的模型训练与推理，需要改哪些地方，越详细越好。谢谢解答

**Comment by UltralyticsAssistant** [comment]
👋 你好 @666git-sys，感谢您对 Ultralytics 的关注 🚀！我们建议您访问 [文档](https://docs.ultralytics.com/)，其中包含许多 [Python](https://docs.ultralytics.com/usage/python/) 和 [CLI](https://docs.ultralytics.com/usage/cli/) 示例，可能已经解答了您的问题。

如果这是一个 🐛 Bug 报告，请提供一个[最小可复现示例](https://docs.ultralytics.com/help/minimum-reproducible-example/) 以帮助我们进行调试。

如果您有关于自定义训练的 ❓问题，请尽可能提供详细信息，包括数据集的图片示例和训练日志，并确保遵循我们的[最佳训练结果提示](https://docs.ultralytics.com/guides/model-training-tips/)。

📢 加入适合您的 Ultralytics 社区：
- 实时聊天：[Discord](https://discord.com/invite/ultralytics) 🎧  
- 深入讨论：[Discourse](https://community.ultralytics.com/)  
- 分享知识：[Subreddit](https://reddit.com/r/Ultralytics)  

## 升级

请升级到最新的 `ultralytics` 包（包括所有 [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)）以确保您的问题未在最新版本中解决。建议使用 [**Python>=3.8**](https://www.python.org/) 和 [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) 环境：

```bash
pip install -U ultralytics
```

## 环境

您可以在以下任一最新验证环境中运行 YOLO（所有依赖项包括 [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn)、[Python](https://www.python.org/) 和 [PyTorch](https://pytorch.org/) 已预安装）：
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. 查看 [GCP 快速上手指南](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. 查看 [AWS 快速上手指南](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker 镜像**. 查看 [Docker 快速上手指南](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## 状态

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

如果此徽章为绿色，则表明所有 [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) 测试当前均已通过。CI 测试每日 24 小时在 macOS、Windows 和 Ubuntu 上验证所有 YOLO [模式](https://docs.ultralytics.com/modes/) 和 [任务](https://docs.ultralytics.com/tasks/)，以及每次提交后的正确运行。

🤖 这是一个自动回复，Ultralytics 工程师将很快为您提供进一步的帮助！

**Comment by Y-T-G** [comment]
Ultralytics supports multiple YOLO models. Not just YOLO11.

https://docs.ultralytics.com/models/yolov8/

**Comment by 666git-sys** [comment]
> Ultralytics 支持多个 YOLO 模型。不仅仅是 YOLO11。
> 
> https://docs.ultralytics.com/models/yolov8/

请问在训练验证推理的过程中如何修改代码呢，除了训练的脚本、推理的脚本、数据集的yaml文件，还需要改动哪些脚本中的代码呢，谢谢解答

**Comment by glenn-jocher** [comment]
Hello! Thanks for reaching out.

Using a YOLOv8 model within the Ultralytics framework is straightforward. You typically only need to specify the YOLOv8 model file (e.g., `yolov8n.pt`) in your Python script or CLI command. The framework is designed to handle different YOLO versions seamlessly without requiring manual code changes in the core library for standard training, validation, or prediction tasks.

For example:

**Python:**
```python
from ultralytics import YOLO

# Load a YOLOv8 model
model = YOLO("yolov8n.pt") 

# Train using YOLOv8
model.train(data="your_dataset.yaml", epochs=100, imgsz=640) 

# Predict using YOLOv8
model.predict(source="your_image.jpg") 
```

**CLI:**
```bash
# Train using YOLOv8
yolo train model=yolov8n.pt data=your_dataset.yaml epochs=100 imgsz=640

# Predict using YOLOv8
yolo predict model=yolov8n.pt source=your_image.jpg
```

You can find more examples in the [Ultralytics YOLOv8 documentation](https://docs.ultralytics.com/models/yolov8/). No other script modifications are usually needed unless you are implementing custom functionalities.

**Comment by 666git-sys** [comment]
> 你好！感谢您的联系。
> 
> 在 Ultralytics 框架中使用 YOLOv8 模型非常简单。通常，您只需在 Python 脚本或 CLI 命令中指定 YOLOv8 模型文件（例如 ）。该框架旨在无缝处理不同的 YOLO 版本，而无需在核心库中手动更改代码以执行标准训练、验证或预测任务。`yolov8n.pt`
> 
> 例如：
> 
> **蟒：**
> 
> from ultralytics import YOLO
> 
> # Load a YOLOv8 model
> model = YOLO("yolov8n.pt") 
> 
> # Train using YOLOv8
> model.train(data="your_dataset.yaml", epochs=100, imgsz=640) 
> 
> # Predict using YOLOv8
> model.predict(source="your_image.jpg") 
> **命令行界面：**
> 
> # Train using YOLOv8
> yolo train model=yolov8n.pt data=your_dataset.yaml epochs=100 imgsz=640
> 
> # Predict using YOLOv8
> yolo predict model=yolov8n.pt source=your_image.jpg
> 您可以在 [Ultralytics YOLOv8 文档](https://docs.ultralytics.com/models/yolov8/)中找到更多示例。除非您正在实现自定义功能，否则通常不需要对脚本进行其他修改。

需要改 head.py 文件中的legacy 这个参数设置为true吗？如果设置了为true后训练和推理的时候会有啥影响吗？

**Comment by Y-T-G** [comment]
You don't have to do anything else. Just follow the docs.

**Comment by wutian-dev** [comment]
Hi, I'd like to ask a question. After I finished training a dataset and generated the YAML file, I found that when using the model to recognize images, only this trained model works independently. My original YOLOv8 model becomes unusable. For example, I trained a model to recognize manhole covers, but this model can only identify manhole covers. When I want to recognize images containing different objects, other objects cannot be detected. Meanwhile, when using the original YOLOv8 model, it cannot recognize manhole covers. How can I add more recognition models to the existing YOLOv8 model, so that my YOLOv8 model can detect more types of objects? All my operations are done in Python; I haven’t used the software mentioned in the lecture.

**Comment by glenn-jocher** [comment]
@wutian-dev When you train a model on a custom dataset (like manhole covers), it learns only those specific classes and "forgets" the original COCO classes it was pre-trained on. To detect both manhole covers and original objects, you need to train on a combined dataset that includes both your custom classes and the original COCO classes, or use separate models for different detection tasks. You can find guidance on [custom training with multiple classes](https://docs.ultralytics.com/modes/train/) in the documentation.
